qadence/ml_tools/train_utils/base_trainer.py

```diff
@@ -330,25 +330,27 @@ def _compute_num_batches(self, dataloader: DataLoader) -> int:
             )
             return min(self.max_batches, n_batches) if self.max_batches is not None else n_batches
 
-    def _validate_dataloader(self, dataloader: DataLoader, dataloader_type: str) -> None:
+    def _validate_dataloader(
+        self, dataloader: DataLoader | DictDataLoader, dataloader_type: str
+    ) -> None:
         """
         Validates the type of the DataLoader and raises errors for unsupported types.
 
         Args:
-            dataloader (DataLoader): The DataLoader to validate.
+            dataloader (DataLoader | DictDataLoader): The DataLoader to validate.
             dataloader_type (str): The type of DataLoader ("train", "val", or "test").
         """
         if dataloader is not None:
-            if not isinstance(dataloader, DataLoader):
+            if not (isinstance(dataloader, DataLoader) or isinstance(dataloader, DictDataLoader)):
                 raise NotImplementedError(
                     f"Unsupported dataloader type: {type(dataloader)}."
                     "The dataloader must be an instance of DataLoader."
                 )
         if dataloader_type == "val" and self.config.val_every > 0:
-            if not isinstance(dataloader, DataLoader):
+            if not (isinstance(dataloader, DataLoader) or isinstance(dataloader, DictDataLoader)):
```

Comment:
Same: `not isinstance(dataloader, (DataLoader, DictDataLoader))`

Reply:
Updated.